laravel/app/Http/Controllers/ScheduleDataController.php

```diff
@@ -579,278 +426,114 @@ private function insertOption()
-                    ->delete();
-                // ->update(['isValid'=>0]);
-            }
-        } else {
-            $overwrittenChoices[2] = 0;
-            if (!is_null($schedule_data3)) {
-                // Insert data
-                Option::insert([
-                    "date" => $date,
-                    "resident" => $resident,
-                    "schedule" => $split[2],
-                    "attending" => $attending,
-                    "option" => $choice,
-                    "milestones" => $_REQUEST["milestones3"],
-                    "objectives" => $_REQUEST["objectives3"],
-                    "anesthesiologist_id" => $pref_anest3,
-                    "isValid" => 1,
-                ]);
-            }
-        }
-
-        // data was overwritten, send a notification
+        // if data was overwritten, send a notification
         if ($notify == true) {
-            // please make sure to change the email here
             self::notifyResidentOverwrittenPreferences(
-                "",
+                $residentName,
                 $_SERVER["HTTP_EMAIL"],
                 $residentName,
                 $date,
                 $overwrittenChoices
             );
         }
+
+        return view("schedules.resident.schedule_update");
+    }
+
+    public function postSubmit($day = null)
+    {
+        self::insertOption();
+        return view("schedules.resident.schedule_update");
     }
 
     public function clearOption($date)
     {
-        // Get resident
         $resident_data = Resident::where("email", $_SERVER["HTTP_EMAIL"])->get();
         $resident = $resident_data[0]["id"];
 
-        //delete first choice data
-        $choice = 1;
-        if (
-            Option::where("date", $date)
-                ->where("resident", $resident)
-                ->where("option", $choice)
-                ->count() != 0
-        ) {
-            Option::where("date", $date)
-                ->where("resident", $resident)
-                ->where("option", $choice)
-                ->delete();
-        }
-        //delete second choice data
-        $choice++;
-        if (
-            Option::where("date", $date)
-                ->where("resident", $resident)
-                ->where("option", $choice)
-                ->count() != 0
-        ) {
-            Option::where("date", $date)
-                ->where("resident", $resident)
-                ->where("option", $choice)
-                ->delete();
-        }
-        //delete third choice data
-        $choice++;
-        if (
-            Option::where("date", $date)
-                ->where("resident", $resident)
-                ->where("option", $choice)
-                ->count() != 0
-        ) {
-            Option::where("date", $date)
-                ->where("resident", $resident)
-                ->where("option", $choice)
-                ->delete();
+        for ($i = 0; $i < count($schedule_data_ids) - 1; $i++)
```

Comment:
is `$schedule_data_ids` defined here?

Reply:
You're right and I'll change it, but unless I'm mistaken, this method never gets called

Reply:
The goal of this method is to delete all the options for a given resident for a given date.

I think it's okay to just do:

```php
Option::where("date", $date)
        ->where("resident", $resident_id)
        ->delete();
```